src/snowflake/snowpark/dataframe_reader.py

```diff
@@ -1489,6 +1494,9 @@ def _read_semi_structured_file(self, path: str, format: str) -> DataFrame:
         df._reader = self
         if xml_reader_udtf:
             set_api_call_source(df, XML_READER_API_SIGNATURE)
+            if self._cur_options.get("CACHERESULT", True):
```

Comment:
Is there a place where we upper case all incoming options?

Reply:
yeah https://github.com/snowflakedb/snowpark-python/blob/f30f60ded986eb2eb531be8d4b2b24dc2291e4f2/src/snowflake/snowpark/dataframe_reader.py#L1107

https://github.com/snowflakedb/snowpark-python/blob/f30f60ded986eb2eb531be8d4b2b24dc2291e4f2/src/snowflake/snowpark/_internal/utils.py#L1339-L1340